lte/gateway/c/oai/tasks/nas/esm/EpsBearerContextDeactivation.c

```diff
@@ -443,50 +446,52 @@ static void _eps_bearer_deactivate_t3495_handler(void *args)
        * The maximum number of deactivate EPS bearer context request
        * message retransmission has exceed
        */
-      pdn_cid_t pid = MAX_APN_PER_UE;
-      int bid = BEARERS_PER_UE;
-
-      /*
-       * Deactivate the EPS bearer context locally without peer-to-peer
-       * * * * signalling between the UE and the MME
-       */
-      rc = _eps_bearer_release(
-        esm_ebr_timer_data->ctx, esm_ebr_timer_data->ebi, &pid, &bid);
-
-      if (rc != RETURNerror) {
-        /*
-         * Stop timer T3495
-         */
-        rc =
-          esm_ebr_stop_timer(esm_ebr_timer_data->ctx, esm_ebr_timer_data->ebi);
+      mme_app_desc_t *mme_app_desc_p = get_mme_nas_state(false);
+      ue_mm_context_t *ue_mm_context = mme_ue_context_exists_mme_ue_s1ap_id(
+        &mme_app_desc_p->mme_ue_contexts, esm_ebr_timer_data->ue_id);
+
+      ebi = esm_ebr_timer_data->ebi;
+      ue_id = esm_ebr_timer_data->ue_id;
+      // Find the index in which the bearer id is stored
+      for (bid = 0; bid < BEARERS_PER_UE; (bid)++) {
+        if (ue_mm_context->bearer_contexts[bid]) {
+          if (ue_mm_context->bearer_contexts[bid]->ebi == ebi) {
+            break;
+          }
+        }
       }
+      // Fetch pdn id using bearer index
+      pdn_cid_t pid = ue_mm_context->bearer_contexts[bid]->pdn_cx_id;
```

Comment:
Use pdn_id instead of pid
Also, add sanity check on "bid" before accessing bearer_contexts[bid]
Check if bid < BEARERS_PER_UE

Reply:
done